core/build.gradle.kts

```diff
@@ -140,6 +141,36 @@ kotlin {
     }
 }
 
+dokka {
+    pluginsConfiguration.html {
+        templatesDir.set(projectDir.resolve("dokka-templates"))
+    }
+
+    dokkaPublications.html {
+        failOnWarning.set(true)
+        // Enum members and undocumented toString()
+        suppressInheritedMembers.set(true)
+    }
+
+    dokkaSourceSets.configureEach {
+        val platform = name.dropLast(4)
+        samples.from("$platform/test")
+
+        // reportUndocumented.set(true) // much noisy output about `hashCode` and serializer encoders, decoders etc
+        skipDeprecated.set(true)
+        // hide the `internal` package, which, on JS, has public members generated by Dukat that would get mentioned
+        perPackageOption {
+            matchingRegex.set(".*\\.internal(\\..*)?")
+            suppress.set(true)
+        }
+        sourceLink {
+            localDirectory.set(rootDir)
+            remoteUrl("https://github.com/Kotlin/kotlinx.collections.immutable/tree/v0.4.0")
```

Comment:
Is there a reason pointing to a particular release instead of the main branch's head?